dymos/utils/introspection.py

```diff
@@ -375,21 +383,70 @@ def configure_parameters_introspection(parameter_options, ode):
-                                                                         user_shape=options['shape'],
-                                                                         user_static_target=options['static_target'])
+            targets = _get_targets_metadata(ode, name=name, user_targets=options['targets'])
         except ValueError as e:
             raise ValueError(f'Parameter `{name}` has invalid target(s).\n{str(e)}') from e
-        options['targets'] = targets
-        options['units'] = units
-        options['shape'] = shape
-        options['static_target'] = static_target
+
+        options['targets'] = list(targets.keys())
+
+        static_tagged_targets = {tgt for tgt, meta in targets.items() if 'dymos.static_target' in meta['tags']}
+
+        # This is a bit of a hack. Any target with a shape of (1,) is unambiguously static.
+        # We may want to consider forcing users to tag these as static for dymos 2.0.0
+        shape_1_targets = {tgt for tgt, meta in targets.items() if meta['shape'] == (1,)}
+        if options['static_targets'] is _unspecified:
+            options['static_targets'] = static_tagged_targets.union(shape_1_targets)
+        elif options['static_targets']:
+            options['static_targets'] = options['targets'].copy()
+        else:
+            options['static_targets'] = []
+
+        if static_tagged_targets and not options['static_targets']:
+            raise ValueError(f"Parameter `{name}` has invalid target(s).\n"
+                             f"User has specified 'static_target = False' for parameter `{name}`,\nbut one or more "
+                             f"targets is tagged with 'dymos.static_target':\n{static_tagged_targets}")
+
+        if options['units'] is _unspecified:
+            options['units'] = _get_common_metadata(targets, metadata_key='units')
+
+        # Check that all targets have the same shape.
+        static_shapes = {}
+        dynamic_shapes = {}
+        # First find the shapes of the static targets
+        for tgt, meta in targets.items():
+            if tgt in options['static_targets']:
+                static_shapes[tgt] = meta['shape']
+            else:
+                if len(meta['shape']) == 1:
+                    dynamic_shapes[tgt] = (1,)
+                else:
+                    dynamic_shapes[tgt] = meta['shape'][1:]
+        all_shapes = {**dynamic_shapes, **static_shapes}
+        # Check that they're unique
+        if len(set(all_shapes.values())) > 1:
+            raise RuntimeError(f'Invalid targets for parameter `{name}`.\n'
+                               f'Targets have multiple shapes.\n'
+                               f'{all_shapes}')
+        elif len(set(all_shapes.values())) == 1:
+            introspected_shape = next(iter(set(all_shapes.values())))
+        else:
+            introspected_shape = None
+
+        if options['shape'] in {_unspecified, None}:
```

Comment:
You use `{_unspecified, None}` or `{None, _unspecified}` in quite a few places in dymos.  Would it be worth it to define a global constant with that value to avoid creating all of those temporary sets?